lib/timeout.rb

```diff
@@ -44,12 +44,43 @@ def self.handle_timeout(message) # :nodoc:
   end
 
   # :stopdoc:
-  CONDVAR = ConditionVariable.new
-  QUEUE = Queue.new
-  QUEUE_MUTEX = Mutex.new
-  TIMEOUT_THREAD_MUTEX = Mutex.new
-  @timeout_thread = nil
-  private_constant :CONDVAR, :QUEUE, :QUEUE_MUTEX, :TIMEOUT_THREAD_MUTEX
+  DATA = Struct.new(*%i{condvar queue queue_mutex timeout_thread_mutex timeout_thread})
+  private_constant :DATA
+
+  private module_function def make_data
+    DATA.new(ConditionVariable.new, # condvar
+             Queue.new, # queue
+             Mutex.new, # queue_mutex
+             Mutex.new, # timeout_thread_mutex
+             nil) # timeout_thread
+  end
+
+  # We keep a private reference so that time mocking libraries won't break
+  # Timeout.
+  begin
+    GET_TIME = Ractor.make_shareable(Process.method(:clock_gettime))
+  rescue
+    GET_TIME = Process.method(:clock_gettime)
+  end
```

Comment:
Could you change it so there is no exception (visible with `-d` and costly on require) when `!defined?(Ractor.make_shareable)`?